avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroValue.scala

```diff
@@ -0,0 +1,63 @@
+package com.sksamuel.avro4s
+
+import java.nio.ByteBuffer
+
+import org.apache.avro.generic.{GenericEnumSymbol, GenericFixed, GenericRecord}
+import org.apache.avro.util.Utf8
+
+sealed trait AvroValue
+
+object AvroValue {
+
+  import scala.collection.JavaConverters._
+
+  case class AvroString(s: String) extends AvroValue
+  case class AvroByteArray(bytes: Array[Byte]) extends AvroValue
+  case class AvroGenericFixed(fixed: GenericFixed) extends AvroValue
+  case class AvroBoolean(boolean: Boolean) extends AvroValue
+  case class AvroShort(short: Short) extends AvroValue
+  case class AvroByte(byte: Byte) extends AvroValue
+  case class AvroDouble(double: Double) extends AvroValue
+  case class AvroFloat(float: Float) extends AvroValue
+  case class AvroInt(int: Int) extends AvroValue
+  case class AvroLong(long: Long) extends AvroValue
+  case class AvroEnumSymbol(symbol: GenericEnumSymbol[_]) extends AvroValue
+  case class AvroList(list: List[AvroValue]) extends AvroValue
+  case class AvroMap(map: Map[String, AvroValue]) extends AvroValue
```

Comment:
Like for lists, I think it would be better to avoid intermediate collections here.

Reply:
And just wrap the raw Java map ?

Reply:
Yes